lib/services/access_request_test.go

```diff
@@ -1869,6 +1869,180 @@ func TestGetResourceDetails(t *testing.T) {
 	require.Equal(t, "friendly group 1", details[types.ResourceIDToString(resourceIDs[4])].FriendlyName)
 }
 
+func TestMaxDuration(t *testing.T) {
+	// describes a collection of roles and their conditions
+	roleDesc := roleTestSet{
+		"requestedRole": {
+			// ...
+		},
+		"requestedRole2": {
+			// ...
+		},
+		"setMaxTTLRole": {
+			options: types.RoleOptions{
+				MaxSessionTTL: types.Duration(8 * time.Hour),
+			},
+		},
+		"defaultRole": {
+			condition: types.RoleConditions{
+				Request: &types.AccessRequestConditions{
+					Roles: []string{"requestedRole", "setMaxTTLRole"},
+				},
+			},
+		},
+		"shortMaxDurationReqRole": {
+			condition: types.RoleConditions{
+				Request: &types.AccessRequestConditions{
+					Roles:       []string{"requestedRole"},
+					MaxDuration: types.Duration(3 * day),
+				},
+			},
+		},
+		"shortMaxDurationReqRole2": {
+			condition: types.RoleConditions{
+				Request: &types.AccessRequestConditions{
+					Roles:       []string{"requestedRole2"},
+					MaxDuration: types.Duration(day),
+				},
+			},
+		},
+	}
+
+	// describes a collection of users with various roles
+	userDesc := map[string][]string{
+		"alice": {"shortMaxDurationReqRole"},
+		"bob":   {"defaultRole"},
+		"carol": {"shortMaxDurationReqRole", "shortMaxDurationReqRole2"},
+	}
+
+	g := getMockGetter(t, roleDesc, userDesc)
+
+	tts := []struct {
+		// desc is a short description of the test scenario (should be unique)
+		desc string
+		// requestor is the name of the requesting user
+		requestor string
+		// the roles to be requested (defaults to "dictator")
+		roles []string
+		// maxDuration is the requested maxDuration duration
+		maxDuration time.Duration
+		// expectedAccessDuration is the expected access duration
+		expectedAccessDuration time.Duration
+	}{
+		{
+			desc:                   "role maxDuration is respected",
+			requestor:              "alice",
+			roles:                  []string{"requestedRole"},
+			maxDuration:            7 * day,
+			expectedAccessDuration: 3 * day,
+		},
+		{
+			desc:                   "maxDuration not set, default maxTTL (8h)",
+			requestor:              "bob",
+			roles:                  []string{"requestedRole"},
+			expectedAccessDuration: 8 * time.Hour,
+		},
+		{
+			desc:                   "maxDuration inside request is respected",
+			requestor:              "bob",
+			roles:                  []string{"requestedRole"},
+			maxDuration:            5 * time.Hour,
+			expectedAccessDuration: 5 * time.Hour,
+		},
```

Comment:
(related to above comment)

Lets add another test case like this, but where bob tries to request more time than he should be permitted without any `MaxDuration` option set for his role, to verify that the "old" (session ttl based) limits are still enforced by default.  Ex:

```golang
{
	desc:                   "users with no MaxDuration are constrained by normal maxTTL logic",
	requestor:              "bob",
	roles:                  []string{"requestedRole"},
	maxDuration:            2 * day,
	expectedAccessDuration: 8 * time.Hour,
},
```